v2/internal/genericarmclient/generic_client_test.go

```diff
@@ -36,7 +36,7 @@ func Test_NewResourceGroup(t *testing.T) {
 	cfg, err := config.ReadFromEnvironment()
 	g.Expect(err).ToNot(HaveOccurred())
 
-	testContext, err := testContext.ForTest(t, cfg)
+	testContext, err := testContext.ForTest(t, cfg, nil)
```

Comment:
TODO: Add a test

Reply:
Forcing consumers to pass a nil value for things they aren't interested in is a bad API design - especially given most consumers won't care.